dags/utils/amplitude.py

```diff
@@ -0,0 +1,133 @@
+from airflow.contrib.operators.bigquery_operator import BigQueryOperator
+from airflow.contrib.operators.bigquery_to_gcs import BigQueryToCloudStorageOperator
+from airflow.contrib.operators.gcs_to_s3 import GoogleCloudStorageToS3Operator
+from operators.bq_sensor import BigQuerySQLSensorOperator
+from operators.gcs import GoogleCloudStorageDeleteOperator
+from os import environ
+
+
+def export_to_amplitude(
+        parent_dag_name,
+        dag_name,
+        default_args,
+        dataset,
+        table_or_view,
+        s3_prefix,
+        gcs_bucket='moz-fx-data-derived-datasets-amplitude-export',
+        gcp_conn_id='google_cloud_derived_datasets',
+        amplitude_s3_conn='amplitude_s3_conn',
+        amplitude_s3_bucket='com-amplitude-vacuum-mozilla-vacuum-wup'):
+
+    """ Export a bigquery table or view to Amplitude.
+
+    This uses the BigQueryToCloudStorage operator to export the
+    partition to GCS, then pushes that data to S3. It operates
+    on a temporary table that is dropped after the job is finished.
+
+    :param str parent_dag_name: Parent dag name
+    :param str dag_name: This dag's name (appended to parent_dag_name)
+    :param str default_args: DAG configuration
+    :param str dataset: BigQuery dataset
+    :param str table_or_view: Table or view name
+    :param str gcs_bucket: The bucket the data will be exported to
+    :param str gcp_conn_id: GCP connection ID
+    :param str amplitude_s3_conn: S3 connection ID
+    :param str amplitude_s3_bucket: The bucket to export data to
+    :param str s3_prefix: The prefix for the s3 objects
+    """
+
+    environment = environ['DEPLOY_ENVIRONMENT']
+    _dag_name = '{}.{}'.format(parent_dag_name, dag_name)
+
+    with models.DAG(_dag_name, default_args=default_args) as dag:
+        # For now, we assume the view is already updated
+        # See https://github.com/mozilla/bigquery-etl/issues/218
+
+        exec_date = '{{ ds }}'
+
+        # Check that we have data for this date
+        check_sql = (
+            'SELECT COUNT(*) '
+            'FROM {}.{} '
+            'WHERE DATE(submission_timestamp) = "{}"'
+        ).format(dataset, table_or_view, exec_date)
+
+        wait_for_data = BigQuerySQLSensorOperator(
+                task_id='wait_for_data',
+                sql=check_sql,
+                bigquery_conn_id=gcp_conn_id,
+                use_legacy_sql=False
+        )
+
+        # Create the table with yesterday's data
+        project_id = GoogleCloudBaseHook(gcp_conn_id=gcp_conn_id).project_id
+        temp_table_name = table_or_view + '_{{ ds_nodash }}'
+        fully_qualified_table_name = '{}.{}.{}'.format(project_id, dataset, temp_table_name)
+
+        sql = (
+            'SELECT * EXCEPT (submission_timestamp) '
+            'FROM {}.{} '
+            'WHERE DATE(submission_timestamp) = "{}"'
+        ).format(dataset, table_or_view, exec_date)
+
+        create_table = BigQueryOperator(
```

Comment:
If no destination table is specified, then BQ will create a temp table to hold query results and that table gets automatically cleaned up after 24 hours. Does the BigQueryOperator give us any way to access the result of the query job in order to get the name of the auto-created table where the results live? If so, that could avoid the need for generating a name here and for cleaning up the table afterward.

Reply:
This seems interesting - let me check.

Reply:
Too bad, it doesn't write to a temporary table if the results are larger than 10GB. That won't happen for Fenix (right now) but is a risk down the line.